application/data/standardisers/ethnicity_classification_finder.py

```diff
@@ -0,0 +1,287 @@
+            classification.get_outputs(raw_ethnicities, self.standardiser) for classification in valid_classifications
+        ]
+        custom_data = EthnicityClassification.get_custom_data_outputs(raw_ethnicities)
+
+        all_output_data = classification_data + [custom_data]
+
+        return all_output_data
+
+    @staticmethod
+    def find_classifications_for_default_finder(raw_ethnicities):
+        custom_data = EthnicityClassification.get_custom_data_outputs(raw_ethnicities)
+        return [custom_data]
+
+    def __get_valid_classifications(self, raw_ethnicities):
+        return self.classification_collection.get_valid_classifications(raw_ethnicities, self.standardiser)
+
+
+class EthnicityStandardiser:
+    def __init__(self, ethnicity_map=None):
+        if ethnicity_map:
+            self.ethnicity_map = ethnicity_map
+        else:
+            self.ethnicity_map = {}
+
+    def __len__(self):
+        return len(self.ethnicity_map)
+
+    def add_conversion(self, raw_ethnicity, standard_ethnicity):
+        lookup_value = self.simplify_key(raw_ethnicity)
+        self.ethnicity_map[lookup_value] = standard_ethnicity
+
+    @classmethod
+    def simplify_key(cls, key_value):
+        return key_value.strip().lower()
+
+    def standardise(self, raw_ethnicity):
+        lookup_value = self.simplify_key(raw_ethnicity)
+        if lookup_value in self.ethnicity_map:
+            return self.ethnicity_map[lookup_value]
+        else:
+            return raw_ethnicity
+
+    def standardise_all(self, raw_ethnicity_list):
+        return [self.standardise(raw_ethnicity) for raw_ethnicity in raw_ethnicity_list]
+
+
+class EthnicityClassificationCollection:
+    def __init__(self):
+        self.classifications = []
+
+    def add_classification(self, classification):
+        self.classifications.append(classification)
+
+    def get_valid_classifications(self, raw_ethnicity_list, ethnicity_standardiser):
+        valid_classifications = [
+            classification
+            for classification in self.classifications
+            if classification.is_valid_for_raw_ethnicities(raw_ethnicity_list, ethnicity_standardiser)
+        ]
+        valid_classifications.sort(
+            key=lambda classification: -classification.get_data_fit_level(raw_ethnicity_list, ethnicity_standardiser)
+        )
+        return valid_classifications
+
+
+class EthnicityClassificationDataItem:
+    """
+    An ethnicity classification data item contains the return data for that
+    """
+
+    def __init__(self, display_ethnicity, parent, order, required):
+        self.display_ethnicity = display_ethnicity
+        self.parent = parent
+        self.order = order
+        self.required = required
+
+    def to_dict(self):
+        return {
+            "display_ethnicity": self.display_ethnicity,
+            "parent": self.parent,
+            "order": self.order,
+            "required": self.required,
+        }
```

Comment:
If this is just a data structure rather than a class that needs functionality associated with it, maybe a namedtuple would work? It comes with its own `_asdict` method, so don't need to roll our own. The declaration would just be this:

`EthnicityClassificationDataItem = collections.namedtuple('EthnicityClassificationDataItem', ('display_ethnicity', 'parent', 'order', 'required'))`

Reply:
It's an alternative but I'm going to stick with this until there is a reason to avoid it